compiler/rustc_mir_build/src/build/matches/mod.rs

```diff
@@ -1962,6 +1962,16 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
             false,
         );
 
+        // If branch coverage is enabled, record this branch.
+        if let Some(branch_info) = self.coverage_branch_info.as_mut() {
+            branch_info.add_two_way_branch(
```

Comment:
This was done deliberately, but deserves more explanation:

- `visit_coverage_branch_condition` performs some extra preprocessing to replace expressions with their enclosing `!` expression, if any. For if-let and let-else, that isn't necessary, since those conditions can't be negated.
- Performing that preprocessing requires a suitable `ExprId` (or adding the ability to pass `None` to skip it). Calling `add_two_way_branch` avoids the need to figure out what (useless) expression ID to pass in.
- `visit_coverage_branch_condition` has an alternate code path that is used when MC/DC coverage is enabled. At the time I originally wrote this PR, I didn't want ongoing work on MC/DC coverage to interfere with my enhancements to branch coverage.
- I could have added a separate `visit_two_way_branch` method that just delegates to `add_two_way_branch` internally, but to me that seemed like unnecessary noise.

Reply:
All that said, I'm not opposed to changing it. The current approach is the one that seemed best to me at the time.

Reply:
Actually, one of the other reasons was that it avoided some awkward partial borrowing scenarios. But those have since been solved in other ways, so maybe I'll try adding the helper method back to see how that looks now.

Reply:
Please do